libslide/src/evaluator_rules/pattern_matching.rs

```diff
@@ -165,28 +172,26 @@ mod tests {
 
     mod match_rule {
         use super::*;
-        use crate::{parse, scan, ScannerOptions};
+        use crate::{parse_expression, parse_expression_pattern, scan};
 
         macro_rules! match_rule_tests {
             ($($name:ident: $rule:expr => $target:expr => $expected_repls:expr)*) => {
             $(
                 #[test]
                 fn $name() {
-                    let parse = |prog: &str, is_rule: bool| -> Expr {
-                        let mut scanner_options = ScannerOptions::default();
-                        if is_rule {
-                            scanner_options = scanner_options.set_is_var_char(|c| {
-                                c.is_alphabetic() || c == '$' || c == '#' || c == '_'
-                            });
-                        }
-                        match parse(scan(prog, scanner_options)) {
-                            Stmt::Expr(expr) => expr,
+                    let parse_rule = |prog: &str| -> ExprPat {
+                        let (expr, _) = parse_expression_pattern(scan(prog));
+                        expr
+                    };
```

Comment:
do you think theres a better way to write this. Maybe assert and then return. It seems weird but this may be rust

Reply:
Assert what? That there are no errors?

Reply:
the double let just looked awkward to me

Reply:
Oh I see. Well `parse_rule` is a closure to parse a pattern. `let (expr, _)` is to destructure the result of parsing the pattern, which returns an expression and any errors.

Reply:
oh, I completely missed the closure part. I understand now. Nice job on this pr!